src/beacon/drand.rs

```diff
@@ -98,16 +90,22 @@ impl BeaconSchedule {
             prev.round()
         };
 
-        let mut cur = max_round;
-        let mut out = Vec::new();
-        while cur > prev_round {
-            // Push all entries from rounds elapsed since the last chain epoch.
-            let entry = curr_beacon.entry(cur).await?;
-            cur = entry.round() - 1;
-            out.push(entry);
+        // We only ever need one entry after nv22 (FIP-0063)
+        if network_version > NetworkVersion::V21 {
```

Comment:
What if `epoch` is between `PineappleUpgradeHeight` and `PineappleUpgradeHeight + 10`